lib/Mongo/MongoClient.php

```diff
@@ -229,7 +262,10 @@ public function listDBs()
      */
     public function selectCollection($db, $collection)
     {
-        return new MongoCollection($this->selectDB($db), $collection);
+        $collection = new MongoCollection($this->selectDB($db), $collection);
+        $collection->setReadPreference($this->readPreference['type'], $this->readPreference['tagsets']);
```

Comment:
Since the `MongoCollection` object inherits the write concern set on the database I think we should move this logic to the constructor of `MongoCollection` and likewise do the same in the constructor for `MongoDB`. This way the logic is preserved even if a `MongoCollection` or `MongoDB` class are created through different means (e.g. creating the object yourself instead of using the helpers).